src/backend/distributed/executor/multi_real_time_executor.c

```diff
@@ -660,9 +676,16 @@ ManageTaskExecution(Task *task, TaskExecution *taskExecution,
 			int32 connectionId = connectionIdArray[currentIndex];
 			int32 fileDesc = fileDescriptorArray[currentIndex];
 			int closed = -1;
+			uint64 bytesReceived = 0;
 
 			/* copy data from worker node, and write to local file */
-			CopyStatus copyStatus = MultiClientCopyData(connectionId, fileDesc);
+			CopyStatus copyStatus = MultiClientCopyData(connectionId, fileDesc,
+														&bytesReceived);
+
+			if (SubPlanLevel > 0 && bytesReceived != 0)
```

Comment:
The `bytesReceived != 0` looks a bit redundant (the comparison is probably about as expensive as the addition)